### PR TITLE
add GET /api/topics endpoint

### PR DESCRIPTION
**Pull request by i786m**
GET /api/topics

responds with a list of topics

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            .then(({ body }) => {
+                const { topics } = body;
+                topics.forEach((el) => {
+                    expect(typeof el).toBe('object');
+                    expect(el).toHaveProperty("slug"), expect.any(String);
+                    expect(el).toHaveProperty("description"), expect.any(String);
+                });
+            });
+    })
+
```
Comment (Dutton1990) [inline __tests__/app.test.js:36]: I think some of these 200 tests could be combined, nothing wrong with splitting them out though if that's what you prefer!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            });
+        });
+
+    test("200: should respond with an array with all topic objects, each of which will have slug and description properties", () => {
+        return request(app)
+            .get("/api/topics")
+            .expect(200)
+            .then(({ body }) => {
+                const { topics } = body;
+                topics.forEach((el) => {
```
Comment (Dutton1990) [inline __tests__/app.test.js]: I'd use a more descriptive parameter name here than 'el'

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+            const { topics } = body;
+            expect(topics.length).toBe(3);
+            });
+        });
+
+    test("200: should respond with an array with all topic objects, each of which will have slug and description properties", () => {
```
Comment (Dutton1990) [inline __tests__/app.test.js]: Great test descriptions!

File: __tests__/app.test.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            });
+        });
+
+    test('200: should return an array with 3 topic objects', () => { 
+        return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+            const { topics } = body;
+            expect(topics.length).toBe(3);
```
Comment (Dutton1990) [inline __tests__/app.test.js]: Important having the length check, nice!

File: __tests__/app.test.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            .then(()=>{
+                return request(app)
+                    .get("/api/topics")
+                    .expect(500)
+                    .then(({body}) => {
+                        const { msg } = body;
+                        expect(msg).toBe('Server Error');
+                    });
+            })
+
```
Comment (Dutton1990) [inline __tests__/app.test.js:51]: Great job with this even if you didn't need to! 😄

**Review by Dutton1990 — APPROVED** [review]
Great job Imran, you can merge and continue!

File: __tests__/app.test.js (hunk 1 above)
Comment (i786m) [reply __tests__/app.test.js:36]: initially i had it all in one test for the 200  but wasnt sure if this was a good approach 
will this suffice 
`
test("200: should respond with an array with 3 topic objects, each of which will have slug and description properties", () => {
        return request(app)
            .get("/api/topics")
            .expect(200)
            .then(({ body }) => {
                const { topics } = body;
                expect(Array.isArray(topics)).toBe(true)
                expect(topics.length).toBe(3)
                topics.forEach((topic) => {
                    expect(typeof topic).toBe('object');
                    expect(topic).toHaveProperty("slug"), expect.any(String);
                    expect(topic).toHaveProperty("description"), expect.any(String);
                });
            });
    })
`

Comment (i786m) [reply __tests__/app.test.js:36]: image
 screenshot of code above for readability

File: __tests__/app.test.js (hunk 2 above)
Comment (i786m) [reply __tests__/app.test.js]: noted and changed

**Pull request merged**